src/utils/telemetry.ts

```diff
@@ -207,3 +225,62 @@ export const consoleTelemetryExporter: TelemetryExporter = {
     console.debug('Telemetry Count', count);
   },
 };
+
+/**
+ * A {@link TelemetryExporter} which does nothing with telemetry events
+ */
+export const nullTelemetryExporter: TelemetryExporter = {
+  /**
+   * Does nothing with the timespan event
+   */
+  sendSpan: () => {},
+
+  /**
+   * Does nothing with the count event
+   */
+  sendCount: () => {},
+};
+
+/**
+ * Telemetry entry point to initialize the singleton {@link TelemetryProvider} or
+ * get the current {@link TelemetryProvider}
+ */
+export class Telemetry {
+  private static instance: TelemetryProvider | undefined;
+
+  /**
+   * Sets the singleton {@link TelemetryProvider} to the given provider
+   *
+   * @param provider The {@link TelemetryProvider} for collecting and sending telemetry
+   */
+  static setInstance(provider: TelemetryProvider) {
+    Telemetry.instance = provider;
+  }
+
+  /**
+   * Returns a {@link DefaultTelemetryProvider} which sends telemetry to the given
+   * telemetry exporter
+   *
+   * @param exporter The exporter for sending telemetry
+   */
+  static withExporter(exporter: 'console' | TelemetryExporter | undefined) : TelemetryProvider {
+    if (exporter === 'console') {
+      return new DefaultTelemetryProvider(consoleTelemetryExporter);
+    }
+    if (exporter && typeof exporter !== 'string') {
+      return new DefaultTelemetryProvider(exporter);
+    }
+    return new DefaultTelemetryProvider(nullTelemetryExporter);
+  }
+
+  /**
+   * Gets the configured {@link TelemetryProvider} instance. Initializes a default
+   * telemetry provider if a telemetry provider hasn't been set
+   */
+  static getInstance(): TelemetryProvider {
+    if (!Telemetry.instance) {
+      Telemetry.instance = Telemetry.withExporter(nullTelemetryExporter);
```

Comment:
This still has a side effect, but I wanted to make it safe to call this method and get a default provider rather than throwing.